### ISSUE TITLE
Run widgets on startup + laggy movements

### ISSUE THREAD

**Issue by Icaras12x** [beyluta/WinWidgets#36]
First of all, brilliant.
Second, I'm wondering why there's no option for "run widgets on startup". I mean, I could write differents widgets and place them all around my desktop in a specific way, but then on a new startup I can't get them to be loaded automatically?

Third, I'm having some issues with lag while moving the widget. My pc is quite medium, ryzen 5 + 16gb ram but I still don't get why would it lag that much.
This is my code:

`
<div class="tradingview-widget-container">
  <div class="tradingview-widget-container__widget"></div>
  <div class="tradingview-widget-copyright"><a href="https://www.tradingview.com/" rel="noopener nofollow" target="_blank"><span class="blue-text">Track all markets on TradingView</span></a></div>
  <script type="text/javascript" src="https://s3.tradingview.com/external-embedding/embed-widget-mini-symbol-overview.js" async>
  {
  "symbol": "FX:EURUSD",
  "width": 350,
  "height": 220,
  "locale": "en",
  "dateRange": "12M",
  "colorTheme": "dark",
  "isTransparent": false,
  "autosize": false,
  "largeChartUrl": ""
}
  </script>
</div>
`

I want to display a tradingview widget with some info about a stock. I'm using this one: https://www.tradingview.com/widget/mini-chart/
Cheers,
Alessio

**Comment by beyluta** [comment]
"Run Widgets on Startup" is a feature I am planning on adding for the next stable release.

And as for the lag, I'm guessing it's due to the following reason:

The logic for mouse up/down detection is handled in the javascript of the widgets to update the positioning of the widget relative to the mouse. The JavaScript is too slow to send the update instructions back to the widget, so it lags when moving the widgets around.

```c#
        private void OnBrowserInitialized(object sender, EventArgs e)
        {
            /*
            @@  Injecting JavaScript code to tell when the window is being dragged.
            */
            browser.ExecuteScriptAsync(@"
                window.onload = () => {
                    let isDrag = false;

                    document.body.onmousedown = (e) => {
                        if (e.buttons === 1) {
                            isDrag = true;
                        }
                    }

                    document.body.onmouseup = () => {
                         isDrag = false;
                    }

                    document.body.onmousemove = () => {
                        if (isDrag) {
                            CefSharp.PostMessage('mouseDrag');
                        }
                    }
                }
            ");

            browser.JavascriptMessageReceived += OnBrowserMessageReceived;
        }
```

The real fix would be to handle this in the form itself instead of the widget's javascript. I tried to fix it once but had issues doing so because the browser element was obstructing the form control. I will attempt to fix the issue again for the next release.

**Comment by beyluta** [comment]
@Icaras12x 

I implemented the "Run Widgets on Startup" and attempted to fix the window move bug (altough it doesn't work the same now; toggling to move will make the widget follow your cursor until you left click).

Could you take a look at these features and tell me if they work? You can find them in the release page of the github under pre-release.

P.S: There are some stability issues with this version so it might crash from time to time. I am working on it. 💪 

https://github.com/beyluta/WinWidgets/releases/tag/v1.3.0-alpha

**Comment by beyluta** [comment]
Btw you have to enable one setting in the settings page of the app. The best way to test it is probably to quit the app through task manager and reopen the program. You can also quit the app through the little icon in the system tray.

![image](https://github.com/beyluta/WinWidgets/assets/5099544/0d00a50a-7d16-47b1-b643-938f20edec57)

**Comment by beyluta** [comment]
should be fixed with #41
